### PR TITLE
Create rule S5868[javascript]: Unicode Grapheme Clusters should be av…

### PR DESCRIPTION
**Pull request by saberduck**
…oided inside regex character classes

### CODE REVIEW

File: rules/S5868/java/rule.adoc [hunk 1]
```diff
@@ -1,10 +1,4 @@
-When placing Unicode https://unicode.org/glossary/#grapheme_cluster[Grapheme Clusters] (characters which require to be encoded in multiple https://unicode.org/glossary/#code_point[Code Points]) inside a character class of a regular expression, this will likely lead to unintended behavior.
-
-
-For instance, the grapheme cluster ``++c̈++`` requires two code points: one for ``++'c'++``, followed by one for the _umlaut_ modifier ``++'\u{0308}'++``. If placed within a character class, such as ``++[c̈]++``, the regex will consider the character class being the enumeration ``++[c\u{0308}]++`` instead. It will, therefore, match every ``++'c'++`` and every _umlaut_ that isn't expressed as a single codepoint, which is extremely unlikely to be the intended behavior.
-
-
-This rule raises an issue every time Unicode Grapheme Clusters are used within a character class of a regular expression.
+include::../description.adoc
```
Comment (yassin-kammoun-sonarsource) [inline rules/S5868/java/rule.adoc:1]:
```suggestion
include::../description.adoc[]
```

File: rules/S5868/javascript/rule.adoc [hunk 2]
```diff
@@ -0,0 +1,18 @@
+include::../description.adoc
```
Comment (yassin-kammoun-sonarsource) [inline rules/S5868/javascript/rule.adoc:1]:
```suggestion
include::../description.adoc[]
```

**Review by yassin-kammoun-sonarsource — CHANGES_REQUESTED** [review]
The rule description is broken as you can preview it [here](https://sonarsource.github.io/rspec/#/rspec/S5868).

**Review by yassin-kammoun-sonarsource — APPROVED** [review]

File: rules/S5868/javascript/rule.adoc [hunk 3]
```diff
@@ -0,0 +1,17 @@
+include::../description.adoc[]
+
+
+== Noncompliant Code Example
+
+----
+/^[Á]$/u
```
Comment (vilchik-elena) [inline rules/S5868/javascript/rule.adoc:7]: I don't get these examples, what are they illustrating? Compliant solution uses different regex. I see that those are examples from ESLint, but I believe it's better to simply adapt examples from Java

**Review by vilchik-elena — DISMISSED** [review]

File: rules/S5868/java/rule.adoc [hunk 4]
```diff
@@ -1,10 +1,4 @@
-When placing Unicode https://unicode.org/glossary/#grapheme_cluster[Grapheme Clusters] (characters which require to be encoded in multiple https://unicode.org/glossary/#code_point[Code Points]) inside a character class of a regular expression, this will likely lead to unintended behavior.
-
-
-For instance, the grapheme cluster ``++c̈++`` requires two code points: one for ``++'c'++``, followed by one for the _umlaut_ modifier ``++'\u{0308}'++``. If placed within a character class, such as ``++[c̈]++``, the regex will consider the character class being the enumeration ``++[c\u{0308}]++`` instead. It will, therefore, match every ``++'c'++`` and every _umlaut_ that isn't expressed as a single codepoint, which is extremely unlikely to be the intended behavior.
-
-
-This rule raises an issue every time Unicode Grapheme Clusters are used within a character class of a regular expression.
+include::../description.adoc[]
```
Comment (vilchik-elena) [inline rules/S5868/java/rule.adoc:1]: `../description.adoc` is empty

**Pull request merged**